enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,532 @@
+"runtimeInfo": {
+  "os": "yqx1WBefoIAq",
+  "osVersion": "AQN-qh_OyRUA",
+  "kind": "wbpgzhNYZQOi",
+  "kindVersion": "cAsoUDy8JFNV",
+  "kindImplementer": "DzP5-GS6tDOQ",
+  "runtimes": [
+    {
+      "name": "zBVuVhrrC_vC",
+      "version": "lyAE7Oh63WPU"
+    }
+  ]
+}
+```
+
+As the values are derived by the container scanner or correspond to software versions, it is possible to create a dictionary of the hashes and retrieve the unobfuscated values. The example above corresponds to:
+
+```json
+"runtimeInfo": {
+  "os": "rhel",
+  "osVersion": "8.9",
+  "kind": "Java",
+  "kindVersion": "17.0.7",
+  "kindImplementer": "Red Hat, Inc.",
+  "runtimes": [
+    {
+      "name": "Quarkus",
+      "version": "2.13.8.Final-redhat-00004"
+    }
+  ]
+}
+```
+
+### Workflow Description
+
+In its existing implementation, the OpenShift Insights Operator periodically gathers workload data every 12 hours.
+
+The Operator iterates over containers in the cluster (up to a limit of 8000) and captures their Image ID and entry points.
+
+When the operator starts gathering workload information, it would first gather runtime information for all the containers running on the cluster's worker nodes and then
+populate the existing gathered info with these additional runtime information.
+
+![container scannner workflow](./container-scanner-diagram-1.png "Container Scanner Worklow")
+
+This is not a user-facing feature. It is achieved within the OpenShift Insights Operator gathering workflow.
+The user is able to examine the gathered data by unarchiving the archive generated by the operator.
+The DaemonSet for the container scanner is deployed on-demand when the operator gathers workload data and is not accessible outside of the `openshift-insights` namespace.
+
+From the operator perspective, the Container Scanner is a black box, and its requirements are:
+
+* Deploy the container scanner `DaemonSet` when gathering workload is starting
+  * The DaemonSet pods are doing nothing when they start. The operator must wait until it is ready and if that takes too long, gather workload without scanning the containers. In that case, an error will be reported by the operator in the archive’s gathers.json file in the `"workloads/workload_info"` section.
+  * The container scanner image must be present in the cluster at installation. The existing tolerations of the Insights Operator should be sufficient to cover the container scanner execution..
+* Determine the Pod name of each container-scanner’s Pod (as the container scanner is deployed as a DaemonSet, there is a single pod instance on each worker node) and map it to its worker node.
+  * There is no service exposing the Container Scanner and its pods are accessed directly.
+* When the operator starts gathering workload information, execute the  `/scan-containers` executable on each container-scanner pod. The executable will output a JSON payload that represents a tree of `namespace/pod-name/container-id/runtime-info`. 
```

Comment:
The output is standard output or?

Reply:
yes, it's standard output. I've clarified it.